sourced/ml/core/algorithms/token_parser.py

```diff
@@ -49,9 +54,17 @@ def __init__(self, stem_threshold=STEM_THRESHOLD, max_token_length=MAX_TOKEN_LEN
         self._single_shot = single_shot
         self._save_token_style = save_token_style
         self._attach_upper = attach_upper
+        self._nn_model = nn_model
+        self._id_splitter_nn = None
+        if use_nn or self._nn_model is not None:
+            self._init_nn()
```

Comment:
I would pass `nn_model` directly to `_init_nn` and remove `self._nn_model` attribute completely.